daffodil-propgen/src/main/resources/org/apache/daffodil/xsd/dfdlx.xsd

```diff
@@ -99,6 +99,7 @@
     <xs:restriction base="xs:string">
       <xs:enumeration value="treatAsEmpty" />
       <xs:enumeration value="treatAsMissing" />
+      <xs:enumeration value="treatAsAbsent" />
```

Comment:
I don't think `treatAsAbsent` should be added here? This simple type restriction is used only for the dfdx extension property, which uses `treatAsMissing` instead of `treatAsAbsent`. If someone wants to use `treatAsAbsent`, let's force them to use the non-extension property.

Reply:
This is a bit of a hack and I couldn't find a way to easily make this cleaner. Let me know if you can think of a better way. 

The issue is that we want to support the new and old attribute, for now, at the same time. An element could have a dfdl:emptyElementParsePolicy or a dfdlx:emptyElementParsePolicy but the property generator fabricates variables using the unprefixed name. Therefore we cannot easily have both of these properties with different types since they map to the same symbol. The schema validator correctly uses the empty/absent enumeration when evaluating the attribute with the dfdl prefix but then the parser uses the dfdlx enumeration definition (which is manually constructed and ignored by the property generator). The absent value was added to the dfdlx enumeration to support both new and old since there can be only one type. I guess the cleanest way would be to change the property generator to use fully qualified attribute names, or at least use them when necessary, but I'm not sure what the wider consequences of that change would be.

Reply:
I think I'm still confused.

I understand why ByHandMixin.scala needs all three enumerations since both dfdl and dfdlx properties use the same EmptyElementParsePolicy object to convert a string to the type.

> but the property generator fabricates variables using the unprefixed name

The property generator excludes the EmptyElementParsePolicy type, so doesn't it completely ignore the definitions in this dfdlx.xsd file? So aren't the enumerations in this file just about doing DFDL schema validation and nothing else? And since treatAsAbsent isn't valid in he dfdlx property shouldn't it be removed here and only be defined in the DFDL_partX xsd file?

Reply:
I think I was the one that was confused. I didn't realize that with the exclusion that the xsd file was completely ignored. I've removed that value from the schema and all the tests run fine. Thanks for that clarification.